### PR TITLE
2610017: update SDK to 1.2.1-beta.1

### PR DESCRIPTION
**Pull request by phileo99**
## Purpose

* Update Calling SDK to v1.2.1-beta.1

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What to Check
Verify that the following are valid
* Tested that general functionality works

### CODE REVIEW

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java [hunk 1]
```diff
@@ -220,10 +224,11 @@ public Boolean isTranscriptionActive() {
         });
     }
 
-    public CompletableFuture hangupAsync() {
+    public CompletableFuture<Void> hangupAsync() {
         call.removeOnRemoteParticipantsUpdatedListener(this::onParticipantsUpdated);
         call.removeOnStateChangedListener(this::onStateChanged);
-        call.removeOnIsRecordingActiveChangedListener(this::onRecordingChanged);
+        recordingFeature.removeOnIsRecordingActiveChangedListener(this::onRecordingChanged);
+
```
Comment (Gemakk) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java:231]: Should also remove transcriptionFeature listener

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java [hunk 2]
```diff
@@ -220,10 +224,11 @@ public Boolean isTranscriptionActive() {
         });
     }
 
-    public CompletableFuture hangupAsync() {
+    public CompletableFuture<Void> hangupAsync() {
```
Comment (Gemakk) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java:227]: What does adding the <Void> do?

**Review by Gemakk — CHANGES_REQUESTED** [review]

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java (hunk 2 above)
Comment (phileo99) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java:227]: adding `<Void>` resolves a Java compiler warning about `CompletableFuture` not having a parameterized type

**Review by Gemakk — APPROVED** [review]
Tested

**Review by iaulakh — APPROVED** [review]

**Pull request merged**